common-security/src/main/java/co/cask/common/security/authorization/client/DefaultAuthorizationClient.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.common.security.authorization.client;
+
+import co.cask.common.security.authorization.ACLEntry;
+import co.cask.common.security.authorization.ACLStore;
+import co.cask.common.security.authorization.AuthorizationContext;
+import co.cask.common.security.authorization.IdentifiableObject;
+import co.cask.common.security.authorization.ObjectId;
+import co.cask.common.security.authorization.SubjectId;
+import co.cask.common.security.authorization.UnauthorizedException;
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
+import com.google.inject.Inject;
+
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Default implementation of {@link AuthorizationClient} which uses {@link ACLStore}.
+ */
+public class DefaultAuthorizationClient implements AuthorizationClient {
+
+  private final ACLStore aclStore;
+  private final AuthorizationContext context;
+
+  /**
+   * @param aclStore used to set and get {@link ACLEntry}s
+   * @param context provides current user and its groups
+   */
+  @Inject
+  public DefaultAuthorizationClient(ACLStore aclStore, AuthorizationContext context) {
+    this.aclStore = aclStore;
+    this.context = context;
+  }
+
+  @Override
+  public void authorize(Iterable<? extends ObjectId> objects, Iterable<? extends SubjectId> subjects,
+                        Iterable<String> requiredPermissions) throws UnauthorizedException {
+    if (!isAuthorized(objects, subjects, requiredPermissions)) {
+      throw new UnauthorizedException();
+    }
+  }
+
+  @Override
+  public void authorize(ObjectId object, Iterable<? extends SubjectId> subjects,
+                        Iterable<String> requiredPermissions) throws UnauthorizedException {
+    if (!isAuthorized(object, subjects, requiredPermissions)) {
+      throw new UnauthorizedException();
```

Comment:
Should contain a message with details on what was actually checked: object, subjects, required permissions.